classes/controller/FrontController.php

```diff
@@ -519,8 +518,14 @@ public function postProcess()
 
     protected function assignGeneralPurposeVariables()
     {
+        if (Validate::isLoadedObject($this->context->cart)) {
+            $presentedCart = $this->cart_presenter->present($this->context->cart);
+        } else {
+            $presentedCart = $this->cart_presenter->present(new Cart());
+        }
```

Comment:
Just a an inspiration for next time, let's factorize duplicated code :)
```suggestion
        if (Validate::isLoadedObject($this->context->cart)) {
            $cart = $this->context->cart;
        } else {
            $cart = new Cart();
        }
        $templateVars = [
            'cart' => $this->cart_presenter->present($cart),
```

Reply:
I agree, this is better :)